src/main/kotlin/parts/wisdom/arcraider/visualization/GridPane.kt

```diff
@@ -34,7 +34,8 @@ class GridPane(val grid: Grid) : JPanel() {
 
         // Draw the active coordinates on the grid
         grid.coords.forEach {
-            graphics2d.fillRect(xStart + it.x * xSize, yStart + it.y * ySize, xSize, ySize)
+            graphics2d.color = colorMap.get(it.color)
```

Comment:
One way to get around this, is to not use a map at all. Instead, use a `when` with the values we care about and an `else` to catch other values. This has the benefit of working for both `null` values as well as values that aren't in our map (like negative or larger than 9)
```kotlin
fun getColor(colorIndex: Int?) = when (colorIndex) {
        1 -> Color.BLACk
        2 -> Color.PINK
        // etc.
        else -> Color.RAINBOW
}
```

FWIW, `it.color` can't be null because it's defined as non-nullable in `Coord`. However, `colorMap.get` _can_ return null if there isn't a corresponding key in the map.

Reply:
Went with Diego's approach